docs/user-documentation.rst

```diff
@@ -0,0 +1,425 @@
+
+Custom conditions
+-----------------
+
+Switchboard supports custom conditions, allowing application developers to
+adapt switches to their particular needs. Creating a condition typically
+consists of extending `switchboard.conditions.ConditionSet`.
+
+An example: if the application needs to activate switches for visitors from a
+particular country, a custom condition can do the geo lookup on the IP from
+the request and return the country value::
+
+    from switchboard.conditions import ConditionSet, Regex
+    from my_app.geo import country_code_by_addr
+
+    class GeoConditionSet(ConditionSet):
+        countries = Regex()
+
+        def get_namespace(self):
+            return 'geo'
+
+        def get_field_value(self, instance, field_name):
+            if field_name == 'countries':
+                return country_code_by_addr(client_ip())
+
+        def get_group_label(self):
+            return 'Geo'
+
+The first thing in the custom condition is to define the fields that makeup the
+condition. In this case, there is one "countries" field, which is a regex,
+allowing admins to specify criteria like `(US|CA)` (US or Canada). Here are the
+fields supported by Switchboard:
+
+* `switchboard.conditions.Boolean` - used for binary, on/off fields
+* `switchboard.conditions.Choice` - used for multiple choice dropdowns
+* `switchboard.conditions.Range` - used for numeric ranges
+* `switchboard.conditions.Percent` - a special type of range specific to
+  percentages
+* `switchboard.conditions.String` - string matching
+* `switchboard.conditions.Regex` - regex expression matching
+* `switchboard.conditions.BeforeDate` - before a date
+* `switchboard.conditions.OnOrAfterDate` - on or after a date
+
+
+Managing switches
+=================
+
+Switches are managed in the dashboard, which is located at the
+`SWITCHBOARD_ROOT` within the application. The dashboard allows:
+
+* Viewing and searching all switches.
+* Reviewing or auditing a switch's history.
+* Adding, editing, and removing switches.
+* Controlling a switch's status.
+* Setting up condition sets for a switch.
+
+Of all these capabilities, the last two are of the most interest, as the status
+and condition sets determine whether a switch is active.
+
+Statuses
+--------
+
+There are four statuses:
+
+* Inactive - disabled for everyone
+* Selective - active only for matched conditions
+* Inherit - inherit from the parent switch
+* Global - active for everyone
+
+Inactive and global are opposite extremes: the switch is turned on or
+off for everyone. The inherit status is used for `Parent-child switches`_. The
+selective status means that the switch is only active if it passes the
+condition sets.
+
+By default, a switch will be created and set to the inactive status. Typical
+workflow would be to put code using a switch into production. It will be
+autocreated on first reference and thus visible in the dashboard. Once
```

Comment:
May be worth noting that not only must the application be running, but the code snippet that uses the switch must be exercised before the autocreation happens.